File: examples/FlightData_3_Reducer.py
# Flight Data 3 Reducer

def reducer(key, values) :
    total = 0
    count = 0
    data = key.split(',')
    year = data[0]
    month = data[1]
    originID = data[2]
    for value in values :
        total = total + float(value)
        count = count + 1
    average = total / count
    output = month + '/' + year + " " + originID
    Wmr.emit(output, average)

File: examples/FlightData_3_Reducer.java
// Flight Data 3 Reducer

public class Reducer {
    public void reduce(String key, WmrIterator values) {
        int total = 0, count = 0;
        for(String value : values) {
            total += Double.parseDouble(value);
            count++;
        }
        double average = (double) total / (double) count;
        Wmr.emit(key, Double.toString(average));
    }
}